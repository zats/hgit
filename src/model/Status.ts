import { Status as GitStatus } from 'nodegit';

export enum Status {
  Unknown,
  Modified,
  Added,
  Removed,
  Clean,
  Missing,
  NotTracked,
  Ignored,
};

export function statusToShortString(status: Status) {
  switch (status) {
    case Status.Unknown: return '';
    case Status.Modified: return 'M';
    case Status.Added: return 'A';
    case Status.Removed: return 'R';
    case Status.Clean: return 'C';
    case Status.Missing: return '!';
    case Status.NotTracked: return '?';
    case Status.Ignored: return 'I';
  }
}

export function gitStatusToStatus(status: GitStatus.STATUS): Status {
  if (status & GitStatus.STATUS.INDEX_NEW) {
    return Status.Added;
  } else if (status & GitStatus.STATUS.WT_NEW) {
    return Status.NotTracked;
  } else if ((status & GitStatus.STATUS.WT_MODIFIED) || (status & GitStatus.STATUS.INDEX_MODIFIED)) {
    return Status.Modified;
  } else if (status & GitStatus.STATUS.WT_DELETED) {
    return Status.Removed;
  }
  console.error(`Unknown status: ${status}`);
  return Status.Unknown;
}
