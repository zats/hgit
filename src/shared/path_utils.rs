use std::path::PathBuf;

pub fn relative_path(full_path: &PathBuf, root_folder: &PathBuf) -> PathBuf {
    let result = match full_path.strip_prefix(root_folder.as_path()) {
        Ok(path) => path,
        Err(error) => full_path.as_path()
    };
    return PathBuf::from(result);
}