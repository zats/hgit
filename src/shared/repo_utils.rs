use git2::{Error, Repository};

pub fn current_repo() -> Result<Repository, Error> {
    let path = ".";
    return Repository::discover(&path);
}