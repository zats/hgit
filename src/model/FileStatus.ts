import { Status } from './Status';

export class FileStatus {
  repoPath: string;
  relativePath: string;
  status: Status;

  constructor(repoPath: string, relativePath: string, status: Status) {
    this.repoPath = repoPath;
    this.relativePath = relativePath;
    this.status = status;
  }

  static sortFunction(s1: FileStatus, s2: FileStatus) {
    const statusOrder = s1.status - s2.status;
    return statusOrder != 0 ? statusOrder : (s1.repoPath > s2.repoPath ? 1 : -1);
  }
}

