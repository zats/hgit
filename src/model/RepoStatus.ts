import { FileStatus } from './FileStatus';
import { Repository } from 'nodegit';

export class RepoStatus {
  repo: Repository;
  fileStatus: FileStatus[];

  constructor(repo: Repository, fileStatus: FileStatus[]) {
    this.repo = repo;
    this.fileStatus = fileStatus;
  }
}