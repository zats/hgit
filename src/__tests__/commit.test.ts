import { createTemporaryRepository, removeTemporaryRepository, hgit, git } from './shared/setup';
import { TestRepo, TestFile } from './shared/model';

beforeEach(createTemporaryRepository);
afterEach(removeTemporaryRepository);

describe('commit', () => {
  it('Last commit message matches', () => {
    const repo = new TestRepo();
    repo.create();
    repo.stage([
      TestFile.createFile('a.txt', 'a.txt content\n'),
      TestFile.createFile('b.txt', 'b.txt content\n'),
      TestFile.createFile('c.txt', 'c.txt content\n'),
    ]);
    hgit(`commit -m "Initial commit"`);
    expect(git('log --format=%B -n 1 HEAD')).toMatchSnapshot();
  });
})

