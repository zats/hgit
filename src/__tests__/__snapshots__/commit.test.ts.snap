// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commit Last commit message matches 1`] = `"Initial commit"`;
